src/serde_macros.rs

```diff
@@ -17,21 +17,63 @@ macro_rules! serde_impl(
 
         impl<'de> ::serde::Deserialize<'de> for $t {
             fn deserialize<D: ::serde::Deserializer<'de>>(d: D) -> Result<$t, D::Error> {
-                use ::serde::de::Error;
                 use hex::FromHex;
 
                 if d.is_human_readable() {
-                    let sl: String = ::serde::Deserialize::deserialize(d)?;
-                    $t::from_hex(&sl).map_err(D::Error::custom)
+                    struct HexVisitor;
+
+                    impl<'de> ::serde::de::Visitor<'de> for HexVisitor {
+                        type Value = $t;
+
+                        fn expecting(&self, formatter: &mut ::std::fmt::Formatter) -> ::std::fmt::Result {
+                            formatter.write_str("an ASCII hex string")
+                        }
+
+                        fn visit_bytes<E>(self, v: &[u8]) -> Result<Self::Value, E>
```

Comment:
I don't know `serde` well enough, but unless there's something that _requires_ `visit_bytes` we can just implement `visit_str` here?

Reply:
The `String` deserializer implements `visit_bytes`, so I didn't want to break anything depending on that.